### PR TITLE
fix: add rel attribute to clickable tile anchor tag

### LINKED ISSUE #2996: anchor rel tag missing in clickable tile for security
## Detailed description

> Describe in detail the issue you're having.
cds-clickable-tile has href target properties, but it is missing the Security HTML anchor rel when target is set

> Is this a feature request (new component, new icon), a bug, or a general issue?
general issue

> Is this issue related to a specific component?
Clickable Tile

> What did you expect to happen? What happened instead? What would you like to see changed?
rel attribute is added

> What browser are you working in?
Chrome, Firefox, safari

> What version of the Carbon Design System are you using?
latest

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

### PR DESCRIPTION
**Pull request by grishma-shah7**
closes #2996

This PR adds the rel attribute to the anchor tag in the clickable tile for security

Similar security also observed in the anchor tag for overflow menu options component

### CODE REVIEW

**Comment by grishma-shah7** [comment]
I have read the DCO document and I hereby sign the DCO.

File: src/tiles/clickable-tile.component.ts [hunk 1]
```diff
@@ -7,6 +7,11 @@ import {
 } from "@angular/core";
 import { Router } from "@angular/router";
 
+/**
+ * Security HTML anchor rel when target is set
+ */
+const REL = "noreferrer noopener";
```
Comment (yinglejia) [inline src/tiles/clickable-tile.component.ts:13]: We shouldn't have a default like this, it needs to be explicitly passed by caller.

File: src/tiles/clickable-tile.component.ts [hunk 2]
```diff
@@ -34,6 +39,7 @@ import { Router } from "@angular/router";
 		(click)="navigate($event)"
 		[attr.href]="disabled ? null : href"
 		[attr.target]="target"
+		[attr.rel]="rel"
```
Comment (yinglejia) [inline src/tiles/clickable-tile.component.ts:42]: Define `rel` as an input property and use something like `[attr.rel]="rel ? rel : null"`

**Review by yinglejia — CHANGES_REQUESTED** [review]

File: src/tiles/clickable-tile.component.ts (hunk 2 above)
Comment (grishma-shah7) [reply src/tiles/clickable-tile.component.ts]: Sure, have added this
I had thought that having the rel attribute for all links would be helpful for usage when rel is not mentioned

**Review by yinglejia — APPROVED** [review]

**Review by Akshat55 — APPROVED** [review]

**Pull request merged**